nixos/modules/services/networking/go-autoconfig.nix

```
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.go-autoconfig;
+  configFile = pkgs.writeText "go-autoconfig.conf" (generators.toKeyValue { } cfg.settings);
+in {
+  options = {
+    services.go-autoconfig = {
+
+      enable = mkEnableOption "IMAP/SMTP autodiscover feature for mail clients";
+
+      port = mkOption {
+        type = types.port;
+        default = 1323;
+        description = "Port to listen on (use 0 for random port).";
```

```suggestion
        description = mdDoc "Port to listen on (use 0 for random port).";
```
on all other options too. :)